lib/Sema/CSSolver.cpp

```diff
@@ -1297,8 +1298,7 @@ ConstraintSystem::solve(Expr *&expr,
   assert(!solverState && "use solveRec for recursive calls");
 
   // Set up the expression type checker timer.
-  Timer.emplace(expr, TC.getDebugTimeExpressions(),
-                TC.getWarnLongExpressionTypeChecking(), TC.Context);
+  startTimer(expr);
```

Comment:
I wonder if the logic of `startTimer` could just be worked into the constructor for `ExpressionTimer`, which is already pretty highly specific to our use case?

Reply:
Sure, if we give it ConstraintSystem& as an argument, it's going to be `Timer.emplace(expr, *this);` WDYT?

Reply:
SGTM!